### ISSUE TITLE
Ember build failing on update to 3.28.6

### ISSUE THREAD

**Issue by pwhittell** [ember-cli/ember-cli#10581]
Attempting to upgrade from 3.7.x to 3.28.6, when building get the following error from the Bundler

Can't resolve 'inggwebplayer/app' in 'C:\storage\git\repos\inggwebplayer'

Really perplexed by this, the app folder is obviously there.

Attached error dump

Output from `ember version --verbose && npm --version && yarn --version && pnpm --version`:
```
ember-cli: 3.28.6
node: 20.18.0
acorn: 8.12.1
ada: 2.9.0
ares: 1.33.1
base64: 0.5.2
brotli: 1.1.0
cjs_module_lexer: 1.4.1
cldr: 45.0
icu: 75.1
llhttp: 8.1.2
modules: 115
napi: 9
nghttp2: 1.61.0
nghttp3: 0.7.0
ngtcp2: 1.1.0
openssl: 3.0.13+quic
simdutf: 5.5.0
tz: 2024a
undici: 6.19.8
unicode: 15.1
uv: 1.46.0
uvwasi: 0.0.21
v8: 11.3.244.8-node.23
zlib: 1.3.0.1-motley-71660e1
os: win32 x64

NPM 10.8.2

No Yarn on pnpm
[error.dump.e2a2bc15a2b2925b2efc2d8f2162c9bf.log](https://github.com/user-attachments/files/18013222/error.dump.e2a2bc15a2b2925b2efc2d8f2162c9bf.log)

```

**Comment by pwhittell** [comment]
inggwebplayer in the first two cases, no name set in ember-cli-build

**Comment by pwhittell** [comment]
The only stack I have is what I attached 
[error.dump.102fe4a914865b475be79abc5dfef196.log](https://github.com/user-attachments/files/18013348/error.dump.102fe4a914865b475be79abc5dfef196.log)

**Comment by NullVoxPopuli** [comment]
looks like it's all ember-auto-import in the top stack frames.
what version do you have that set at?

**Comment by pwhittell** [comment]
I tried removing the auto-importer dependency and it seems to have built SOMETHING, will see if this works

**Comment by pwhittell** [comment]
Yeah, that build didn't have any dependencies as you might expect

Per the package "ember-auto-import": "^1.12.0",

**Comment by NullVoxPopuli** [comment]
can you try latest v2? (2.10.x I think ?)

ember-auto-import is very important for general happiness in the codebase -- would not recommend getting rid of it :see_no_evil:

**Comment by pwhittell** [comment]
Tried upgrading to the latest ember-auto-import, also reverted the app name to what it was before (tried changing to match the folder name before as an attempt to get out of this hell)

New stack trace attached, it seems to not be able to find Node_Modules, even though it's there
[error.dump.c1a141755e085c6b93559b30b5ecc678.log](https://github.com/user-attachments/files/18013478/error.dump.c1a141755e085c6b93559b30b5ecc678.log)

**Comment by NullVoxPopuli** [comment]
do you have any issues with a blank/fresh 3.28.x project?


> ingg-web-viewer

looks like the same problem -- it doesnt' know what `ingg-web-viewer` -- what are the imports you use in your app? do they match your package.json#name and environment.js#modulePrefix?

**Comment by pwhittell** [comment]
They do. I'm sure a fresh project would be fine, issue is I need to upgrade this one, as Ember has failed a security audit. Any suggestions would be welcome

**Comment by NullVoxPopuli** [comment]
if you diff the folders of your project and a new project, what do you observe? you could even try to generate the new project with the same name -- and use any diff tool to diff the folders.

As for failing security audits, you have quite a few upgrades to go to catch up <3

**Comment by pwhittell** [comment]
When you mentioned the imports, only thing I could think might be an issue is, I believe I ran the code-mod command after the cli upgrade command, but I think this initial command failed due to targeting the wrong version. Could this code mod command have changed some of the import statements so that it's now in this state?

And on the subject of the security audits, can you advise the earliest version I'd have to upgrade to to get away from the issue of prototype pollution?

**Comment by NullVoxPopuli** [comment]
> . Could this code mod command have changed some of the import statements so that it's now in this state?

maybe -- depends on which codemod ran, and what your diff is

> can you advise the earliest version I'd have to upgrade to to get away from the issue of prototype pollution?

I am not sure about that one, but our security policy is here: https://emberjs.com/security/

And our release policy is here: https://emberjs.com/releases/
the LTS policy is what gets the longtail of security fixes: https://emberjs.com/releases/lts/

On the LTS page, the table towards the bottom shows that the oldest supported version of ember-source is 5.4:
![image](https://github.com/user-attachments/assets/8d4d3060-2a50-4aa8-a27a-7107710b581f)

and 4.12 for ember-data
![image](https://github.com/user-attachments/assets/982ebf13-6bdb-4e49-a5c1-08b46dfbf565)

**Comment by pwhittell** [comment]
Attempting to upgrade to 4.10.0, same error, I believe it is the environment.js that is failing to build. We have had to make mods to it in the past to enable some parameterised builds, here is the ENV we are attempting to build, with the codemods etc applied, can you see anything that would break here?

'use strict';
/* eslint-env node */
var argv = require('minimist')(process.argv.slice(0));
var fileSystem = require("fs"); //this gets file system module as a variable https://www.w3schools.com/nodejs/nodejs_filesystem.asp
console.log('Mode is set to ' + argv.mode);
//logging activated?
var isLoggingOn = false;
if (argv.logging == null) {
    console.log("logging parameter not specified, defaulting to no logging.  Use --logging=true to turn on.");
} else {
    isLoggingOn = (argv.logging.toUpperCase() === "true".toUpperCase());
    console.log('logging is set to ' + isLoggingOn);
}
var baseUrlOverride = '/inggWebViewer/';
if (argv.baseUrl == null) {
    console.log("Base URL is not being overridden, will default to inggWebViewer");
}
else {
    baseUrlOverride = argv.baseUrl;
    console.log("Base URL being overridden to " + baseUrlOverride);
}
var tech = 'hls';
if (argv.tech == null) {
    console.log("Player tech is not set, will default to using HLS");
}
else {
    tech = argv.tech;
    console.log("Player tech is set to " + tech);
}

module.exports = function (environment) {
    const ENV = {
        modulePrefix: 'ingg-web-viewer',
        environment,
        VC_mode: argv.mode,
        locationType: 'history',
        isLoggingOn: isLoggingOn,
        rootURL: baseUrlOverride,
        playerTech: tech,
        EmberENV: {
            EXTEND_PROTOTYPES: false,
            FEATURES: {
                // Here you can enable experimental features on an ember canary build
                // e.g. 'with-controller': true
            }
        },

        APP: {
            // Here you can pass flags/options to your application instance
            // when it is created
            pomxml: fileSystem.readFileSync("./pom.xml","utf-8")
        },

        webFontConfig: {
            google: {
                families: ['News Cycle']
            },
            timeout: 2000
        }
    };

    if (environment === 'development') {
        // ENV.APP.LOG_RESOLVER = true;
        // ENV.APP.LOG_ACTIVE_GENERATION = true;
        // ENV.APP.LOG_TRANSITIONS = true;
        // ENV.APP.LOG_TRANSITIONS_INTERNAL = true;
        // ENV.APP.LOG_VIEW_LOOKUPS = true;
    }

    if (environment === 'test') {
        // Testem prefers this...
        ENV.locationType = 'none';
        ENV.rootURL = '/';
        // keep test console output quieter
        ENV.APP.LOG_ACTIVE_GENERATION = false;
        ENV.APP.LOG_VIEW_LOOKUPS = false;

        ENV.APP.rootElement = '#ember-testing';
        ENV.APP.autoboot = false;
    }

    return ENV;
};

**Comment by NullVoxPopuli** [comment]
Can you put that in a codefence? I can't read it. Thanks!

~~~
```js
Code here 
```
~~~

**Comment by NullVoxPopuli** [comment]
Can you make an oss reproduction? I'm not sure what else to look at without

**Comment by pwhittell** [comment]
Probably not, at this stage. We need to consider dropping Ember going forward; too much headache just to have a build system and injected services

**Comment by NullVoxPopuli** [comment]
Alright. Managing dependencies isn't any different in other ecosystems tho :/

I'll close this issue since we've hit a debugging dead end (need a repro (usually the process of creating a replo tells you what needs to be fixed as well))
